code/mathematical_algorithms/src/prime_numbers_of_n/prime_numbers_of_n.c

```diff
@@ -0,0 +1,35 @@
+# include <stdio.h>
+# include <math.h>
+
+void 
+primeFactors(int number)
+{
+    int i;
+    while (number % 2 == 0) {
+        printf("%d ", 2);
+        number = number / 2;
+    }
+
+    for (i = 3; i <= sqrt(number); i = i + 2) {
+        while (number % i == 0) {
+            printf("%d ", i);
+            number = number / i;
+        }
+    }
+
+    if (number > 2)
+        printf ("%d ", number);
```

Comment:
I think returning a value would be better than printing one.

Reply:
The variable "number" is getting modified inside the function multiple times. Printing prime factors is happening at 3 places (line 9,15,21)inside the function.So, I think it is not possible to return the value to main function at line 21. Moreover, the code is more readable if a single function prints all prime factors.

Reply:
@arnavb What do you suggest about it? Is this program fine?

Reply:
I meant returning an array of prime numbers should be better. A function should only have 1 job.